source/val/validate_conversion.cpp

```diff
@@ -473,21 +485,37 @@ spv_result_t ConversionPass(ValidationState_t& _, const Instruction* inst) {
       const bool input_is_pointer = _.IsPointerType(input_type);
       const bool input_is_int_scalar = _.IsIntScalarType(input_type);
 
-      if (!result_is_pointer && !result_is_int_scalar &&
+      const bool result_is_coopmat = _.IsCooperativeMatrixKHRType(result_type) ||
+                                     _.IsCooperativeMatrixNVType(result_type);
+      const bool input_is_coopmat = _.IsCooperativeMatrixKHRType(input_type) ||
+                                    _.IsCooperativeMatrixNVType(input_type);
```

Comment:
Are you testing that both are coop matrices? If so, the generic test should be sufficient.

Reply:
No, the code in lines 493-513 test various combinations of input and output.

I do not see how it can be optimized.

Reply:
Wouldn't just testing `IsCooperativeMatrixType()` be cleaner than testing both flavours?

Reply:
Agree